policy/composer.go

```diff
@@ -192,6 +168,187 @@ func (opt *ruleComposerImpl) registerVariable(ctx *cel.OptimizerContext, v *Comp
 	opt.nextVarIndex++
 }
 
+// sortedVariables returns the variables ordered by their declaration index.
 func (opt *ruleComposerImpl) sortedVariables() []varIndex {
 	return opt.varIndices
 }
+
+// outputStep interface represents a policy output expression.
+type outputStep interface {
```

Comment:
Just reading the description, I don't understand the difference between this and CompiledRule. It seems to me that CompiledRule could also define all the methods in this interface. Could we make the difference explicit? What processing happens to turn a CompiledRule into this?

Reply:
The output step could either be a `CompiledMatch` or a `CompiledRule`, and it represents some intermediate state of expression composition. I've updated the comment to indicate the purpose of this interface. It's likely something I'll elaborate on in further PRs to introduce new composition semantics.